internal/receiver/scriptedinputsreceiver/procpipe/procpipe.go

```diff
@@ -0,0 +1,233 @@
+	if err != nil {
+		return nil, err
+	}
+
+	// If MaxLogSize not set, set sane default
+	if c.MaxLogSize == 0 {
+		c.MaxLogSize = DefaultMaxLogSize
+	}
+
+	if c.CollectionInterval == "" {
+		c.CollectionInterval = strconv.Itoa(DefaultIntervalSeconds)
+	}
+
+	encoding, err := c.Encoding.Build()
+	if err != nil {
+		return nil, err
+	}
+
+	// Build multiline
+	var splitFunc bufio.SplitFunc
+	if c.Multiline.LineStartPattern == "" && c.Multiline.LineEndPattern == "" {
+		splitFunc = helper.SplitNone(int(c.MaxLogSize))
+	} else {
+		splitFunc, err = c.Multiline.Build(encoding.Encoding, true, false, false, nil, int(c.MaxLogSize))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &Input{
+		baseConfig:    c.BaseConfig,
+		InputOperator: inputOperator,
+		logger:        logger,
+		MaxLogSize:    int(c.MaxLogSize),
+		encoding:      encoding,
+		splitFunc:     splitFunc,
+	}, nil
+}
+
+// Input is an operator that reads input from stdin
+type Input struct {
+	logger    *zap.SugaredLogger
+	cancelAll context.CancelFunc
+	splitFunc bufio.SplitFunc
+	helper.InputOperator
+	encoding   helper.Encoding
+	baseConfig BaseConfig
+	wg         sync.WaitGroup
+	MaxLogSize int
+}
+
+func createTicker(intervalStr string) (*time.Ticker, error) {
+	interval, err := time.ParseDuration(intervalStr)
+	if err != nil {
+		return nil, err
+	}
+	return time.NewTicker(interval), nil
+}
+
+// Start will start generating log entries.
+func (g *Input) Start(_ operator.Persister) error {
+
+	ctx, cancelAll := context.WithCancel(context.Background())
+	g.cancelAll = cancelAll
+
+	ticker, err := createTicker(g.baseConfig.CollectionInterval)
+	if err != nil {
+		// TODO: move ticker verification to Start() so that we cannot error here.
+		return err
+	}
+
+	go func() {
+		for {
+			_, cancelCycle := context.WithCancel(ctx)
+
+			err := g.beginCycle(ctx)
+			if err != nil {
+				g.logger.Errorf("Error running script: %v", err)
+			}
+
+			select {
+			case <-ctx.Done():
+				cancelCycle()
+				return
+			case <-ticker.C:
+				cancelCycle()
+				g.wg.Wait()
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (g *Input) beginCycle(ctx context.Context) error {
+
+	stdOutReader, stdOutWriter := io.Pipe()
+	commander, err := NewCommander(g.logger.Desugar(), g.baseConfig.ScriptName, stdOutWriter)
+	if err != nil {
+		return err
+	}
+
+	if err := commander.Start(ctx); err != nil {
+		return err
+	}
+
+	g.wg.Add(2)
+
+	readerCtx, cancelReader := context.WithCancel(ctx)
+
+	go func() {
+		defer g.wg.Done()
+		select {
+		case <-commander.Done():
+			g.logger.Debug("Script finished", zap.String("exec_file", g.baseConfig.ScriptName))
+			// Close the write pipe. This will result in subsequent read by scanner to return EOF and finish
+			// the goroutine that processes the script output.
+			err := stdOutWriter.Close()
+			if err != nil {
+				return
+			}
+
+		case <-ctx.Done():
+			g.logger.Info("Script run too long. Stopping.", zap.String("exec_file", g.baseConfig.ScriptName))
```

Comment:
This is from a context.WithCancel and isn't based on a timer? How is the "Script run too long" determination made?